lib/presentation/forget_password/forget_password_view_model.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart';
+import 'package:online_exam_app/core/base/base_view_model.dart';
+import 'package:online_exam_app/domain/core/results.dart';
+import 'package:online_exam_app/domain/entities/authentication/forgetPassword/forget_password_request.dart';
+import 'package:online_exam_app/domain/entities/authentication/forgetPassword/forget_password_response.dart';
+import 'package:online_exam_app/domain/use_case/forget_password_use_case.dart';
+import 'package:online_exam_app/presentation/forget_password/forget_password_contract.dart';
+
+@injectable
+class ForgetPasswordViewModel extends BaseViewModel<ForgetPasswordViewState> {
+  final ForgetPasswordUseCase _forgetPasswordUseCase;
+
+  ForgetPasswordViewModel(this._forgetPasswordUseCase)
+      : super(InitialForgetPasswordViewState());
+
+  final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+  final TextEditingController emailController = TextEditingController();
+  bool valid = false;
+
+  String? emailValidation(String input) {
+    if (input.isEmpty) {
+      return locale!.emailCantBeEmpty;
+    } else if (!RegExp(r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+"
+            r"@[a-zA-Z0-9](?:[a-zA-Z0-9-]"
+            r"{0,253}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]"
+            r"{0,253}[a-zA-Z0-9])?)*$")
+        .hasMatch(input)) {
+      return locale!.enterAValidEmail;
+    }
+    return null;
+  }
+
+  void doIntent(ForgetPasswordViewAction action) {
+    switch (action) {
+      case FormDataChangedAction():
+        {
+          _updateValidationState();
+        }
+      case ForgetPasswordAction():
+        {
+          _forgetPassword();
+        }
+      case NavigateToOtpScreenAction():
+        {
+          _navigateToOtpScreen();
+        }
+    }
+  }
+
+  void _forgetPassword() async {
+    if (formKey.currentState!.validate()) {
+      emit(ForgetPasswordLoadingState());
+      var response = await _forgetPasswordUseCase
+          .call(ForgetPasswordRequest(email: emailController.text));
+      emit(HideLoadingState());
```

Comment:
replace this code with this 
var response = await _forgetPasswordUseCase(ForgetPasswordRequest(email: emailController.text));